keps/sig-windows/2802-identify-windows-pods-apiserver-admission/README.md

```diff
@@ -0,0 +1,393 @@
+different and not all Kubernetes features are supported on Windows host.
+
+
+### User Stories (Optional)
+
+
+#### Story 1
+
+As a Kubernetes cluster administrator, I want appropriate security contexts to be 
+applied to my Windows Pods along with Linux pods
+
+#### Story 2
+
+As a Kubernetes cluster administrator, I want to use my own admission webhook for pods
+based on the OS it intends to run on.
+
+### Notes/Constraints/Caveats (Optional)
+
+Since the OS field is optional in the pod spec, we can expect the users to omit this field
+when submitting pod to API server. In this scenario, we expect admission plugins to treat the pod
+the way it is being treated now.
+
+### Risks and Mitigations
+
+The primary risk is a bug in the implementation of the admission plugins that validate or mutate
+based on the OS field in the pod spec. The best mitigation for that scenario is unit testing 
+when this featuregate is enabled and disabled.
+
+
+## Design Details
+
+### Implications to Kubelet
+
+Apart from the above API change, we intend to make the following changes to Kubelet:
+- Kubelet should reject admitting pod if the kubelet cannot honor the pod.Spec.OS.Name. For instance, if the OS.Name does not match the host os.
+- Kubelet should reject admitting pods if the pod contains a nodeSelector with label `kubenetes.io/os` with a value 
+not matching `runtime.GOOS`. We can use this check as `kubernetes.io/os` value is supposed to be filled from `runtime.GOOS`. 
+
+Having the above checks in kubelet helps in the following conditions:
+- If scheduler has been by passed by setting the nodeName directly on the pod spec.
+- A malicious user or a user unknowingly with RBAC to just bind pod to node, can associate pod to node there by bypassing the scheduler.
+
+As of now, Kubelet doesn't reconcile the value of `kubernetes.io/os`. So, it'd be nice to have the reconciliation as part of this change.
+
+
+### Implications to Scheduler
+
+We let the users to explicitly specify nodeSelectors+tolerations or runtimeclasses to express their intention
+to run an particular OS. However, in future, once the OS struct expands, we can see if we can leverage those fields to
+express scheduling constraints. During the alpha, we assume there are no scheduling implications.
+
+
+### Test Plan
+
+- Unit tests covering API server defaulting to various fields within pod with and without this feature
+- Unit tests covering admission plugins which validate/mutate pod spec based on this feature
+- Unit and E2E tests for Kubelet changes.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Basic units and e2e tests completed and enabled
+
+#### Alpha -> Beta Graduation
+- Expand the e2e tests with more scenarios
+- Gather feedback from end users
+- Tests are in Testgrid and linked in the KEP
+
+#### Alpha -> Beta Graduation
+- 2 examples of end users using this field
```

Comment:
```suggestion
- end to end tests confirmed as passing, with the new label, on old as well as new APIServer/Kubelet versions (i.e. proof of backwards compatibilty) 
```

Reply:
What label are you referring to here?

Reply:
s/label/thing that we use to identify the host OS/g :)